scripts/apps/search/directives/SearchPanel.ts

```diff
@@ -63,16 +60,24 @@ export function SearchPanel($location,
             scope.isManagingSubscriptions = false;
             scope.wrapper = {};
 
-            scope.manageSubscriptions = (nextValue) => {
+            scope.setIsManagingSubscriptions = (nextValue: boolean) => {
                 scope.isManagingSubscriptions = nextValue;
             };
 
+            // called after changing the subscriptions for current search
+            scope.onSubscriptionsChange = (updatedSavedSearch) => {
+                // subscriptions were updated via API so the object has changed
+                for (const key in updatedSavedSearch) {
+                    scope.editingSearch[key] = updatedSavedSearch[key];
```

Comment:
Why don't update the entire object like `scope.editingSearch = updatedSavedSearch`? Is angular not picking up the updates?

Reply:
For some reason I still don't know, it doesn't. `scope.editingSearch = updatedSavedSearch` doesn't work.

Reply:
I'm pretty sure it's this one https://stackoverflow.com/a/38763654 which seems to be everyone "not understanding scopes" and definitely not an angular bug which is probably hard to solve in the framework and everyone is doing workarounds. :smile:

Reply:
🤦 I hate angular... So basically it's [shadowing the parent variable](https://github.com/angular/angular.js/wiki/Understanding-Scopes). I'll fix it

Reply:
Actually, something else is happening. When I do `scope.editingSearch._etag = `, it's also changing `scope.edit._etag` (which is the etag that matters on the request), but I couldn't find why they are bound together.

Reply:
This doesn't happen when I change all of the `scope.edit = create(scope.editingSearch)` to `scope.edit = scope.editingSearch`. But I'm not sure why lodash `create` would bind those together.